src/encryption.rs

```diff
@@ -108,11 +109,22 @@ impl Encryption<'_> {
 
     fn get_sec_key(&self, key: &str) -> Result<SecretKey, &'static str> {
         let sec_key = self.resolve_keys(key)?.secret_key;
-
-        // sodiumoxide needs fixed-length arrays, not slices
-        let seckey_decoded = BASE64.decode(sec_key.as_bytes()).unwrap_or_default();
-        if seckey_decoded.len() != SECRETKEYBYTES {
-            return Err("The secret key did not match the expected format.");
+        let seckey_decoded: Vec<u8>;
+
+        if sec_key.as_bytes().len() == 64 {
+            seckey_decoded = HEXUPPER_PERMISSIVE
+                .decode(sec_key.as_bytes())
+                .unwrap_or_default();
+
+            if seckey_decoded.len() != SECRETKEYBYTES {
```

Comment:
Can't this be outside of the if-else given it's the same for both?

Reply:
yes